client/v3/concurrency/session.go

```diff
@@ -108,14 +110,16 @@ type sessionOptions struct {
 }
 
 // SessionOption configures Session.
-type SessionOption func(*sessionOptions)
+type SessionOption func(*sessionOptions, *zap.Logger)
 
 // WithTTL configures the session's TTL in seconds.
 // If TTL is <= 0, the default 60 seconds TTL will be used.
 func WithTTL(ttl int) SessionOption {
-	return func(so *sessionOptions) {
+	return func(so *sessionOptions, lg *zap.Logger) {
 		if ttl > 0 {
 			so.ttl = ttl
+		} else {
+			lg.Info("TTL should be > 0, using default", zap.Int64("default-session-ttl", defaultSessionTTL))
```

Comment:
I would even make it "Warn", as this option to reset to default does not have much sense.